contracts/src/v0.8/automation/dev/v2_3/AutomationRegistryBase2_3.sol

```diff
@@ -944,6 +951,12 @@ abstract contract AutomationRegistryBase2_3 is ConfirmedOwner {
     PaymentParams memory paymentParams,
     uint256 upkeepId
   ) internal returns (PaymentReceipt memory) {
+    BillingConfig memory billingConfigOverridden = s_billingOverrides[upkeepId];
+    if (address(billingConfigOverridden.priceFeed) != ZERO_ADDRESS) {
+      // use the overridden configs
+      paymentParams.billingToken.gasFeePPB = billingConfigOverridden.gasFeePPB;
+      paymentParams.billingToken.flatFeeMicroLink = billingConfigOverridden.flatFeeMicroLink;
```

Comment:
@RyanRHall  comparing the BillingConfig struct and this PaymentsParams struct, I think only these 2 configs are override-able. Let me know if anything else to override?

also, since the `_calculatePaymentAmount ` func is upkeepID agnostic, i hope it is okay to override before the _calculatePaymentAmount call.

Reply:
hey sorry this was a bit confusing. I think you could define a new struct, something like...
```sol
struct BillingOverrides {
  gasFeePPB;
  flatFeeMicroLink;  // note this will become flatFeeMilliCents soon
}
```